src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -11449,119 +11557,327 @@ def update_tag(
+                    resource_id=resource.id,
+                    resource_type=resource_type,
+                )
             )
+            session.add(tag_resource_schema)
+            tag_resource_schemas.append(tag_resource_schema)
+
+            # If the tag is an exclusive tag, apply the check and attach/detach accordingly
+            if tag_schema.exclusive:
+                scope_ids: Dict[
+                    TaggableResourceTypes, List[Union[UUID, int]]
+                ] = defaultdict(list)
+                detach_resources: List[TagResourceRequest] = []
+
+                if isinstance(resource, PipelineRunSchema):
+                    if resource.pipeline_id:
+                        scope_ids[TaggableResourceTypes.PIPELINE_RUN].append(
+                            resource.pipeline_id
+                        )
+                    other_runs_with_same_tag = self.list_runs(
+                        PipelineRunFilter(
+                            id=f"notequals:{resource.id}",
+                            pipeline_id=resource.pipeline_id,
+                            tags=[tag_schema.name],
+                        )
+                    )
+                    if other_runs_with_same_tag.items:
+                        detach_resources.append(
+                            TagResourceRequest(
+                                tag_id=tag_schema.id,
+                                resource_id=other_runs_with_same_tag.items[
+                                    0
+                                ].id,
+                                resource_type=TaggableResourceTypes.PIPELINE_RUN,
+                            )
+                        )
+                elif isinstance(resource, ArtifactVersionSchema):
+                    if resource.artifact_id:
+                        scope_ids[
+                            TaggableResourceTypes.ARTIFACT_VERSION
+                        ].append(resource.artifact_id)
+                    other_versions_with_same_tag = self.list_artifact_versions(
+                        ArtifactVersionFilter(
+                            id=f"notequals:{resource.id}",
+                            artifact_id=resource.artifact_id,
+                            tags=[tag_schema.name],
+                        )
+                    )
+                    if other_versions_with_same_tag.items:
+                        detach_resources.append(
+                            TagResourceRequest(
+                                tag_id=tag_schema.id,
+                                resource_id=other_versions_with_same_tag.items[
+                                    0
+                                ].id,
+                                resource_type=TaggableResourceTypes.ARTIFACT_VERSION,
+                            )
+                        )
+                elif isinstance(resource, RunTemplateSchema):
+                    scope_ids[TaggableResourceTypes.RUN_TEMPLATE].append(0)
```

Comment:
Why do we append `0` here?

Reply:
With the new scoping, the `0` is irrelevant now.